src/main/java/org/openbase/bco/bcozy/controller/powerterminal/PowerChartVisualizationController.java

```diff
@@ -0,0 +1,297 @@
+                }
+                chart.addSeries(series);
+                break;
+        }
+        chart.setSkinType(skinType);
+    }
+
+
+    /**
+     * Refreshes all data every Minute
+     * @param refreshTimeout
+     * @param data
+     */
+    private void enableDataRefresh(int refreshTimeout, List<ChartData> data) {
+        try {
+            GlobalScheduledExecutorService.scheduleAtFixedRate(() -> {
+                Platform.runLater(() -> {
+                    setChartType(this.visualizationType);
+                });
+            }, 1, refreshTimeout, TimeUnit.MINUTES);
+        } catch (NotAvailableException ex) {
+            ExceptionPrinter.printHistory("Could not refresh power chart data", ex, LOGGER);
+        }
+    }
+
+
+    /**
+     * Initializes the previous energy consumption
+     *
+     * @return List of ChartData with previous Energy Consumption
+     */
+    private List<ChartData> initializePreviousEntries(String interval, long startTime, long endTime) {
+        List<ChartData> datas = new ArrayList<ChartData>();
+        int change = 0;
+        olddataTime = new Timestamp(TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()));
+        try {
+            List<FluxTable> energy = InfluxDBHandler.getAveragePowerConsumptionTables(
+                    interval, TimeUnit.MILLISECONDS.toSeconds(startTime), TimeUnit.MILLISECONDS.toSeconds(endTime), "consumption");
+            for (FluxTable fluxTable : energy) {
+                List<FluxRecord> records = fluxTable.getRecords();
+                for (FluxRecord fluxRecord : records) {
+                    if (fluxRecord.getValueByKey("_value") == null) {
+                        ChartData temp = new ChartData(String.valueOf(change), 0, Tile.ORANGE);
+                        temp.setName(String.valueOf(change));
+                        datas.add(temp);
+                    } else {
+                        ChartData temp = new ChartData(String.valueOf(change), (double) fluxRecord.getValueByKey("_value") / dataStep, Tile.ORANGE);
+                        temp.setName(String.valueOf(change));
+                        datas.add(temp);
+                    }
+
+                }
+                change++;
+            }
+        } catch (CouldNotPerformException e) {
+            e.printStackTrace();
+        }
+        return datas;
+    }
+
+    /**
+     * Connects the given chart attribute properties to the chart by creating listeners incorporating the changes
+     * into the chart
+     * @param chartStateModel StateModel that describes the state of the chart as configured by other panes
+     */
+    public void initChartState(ChartStateModel chartStateModel) {
+
+        this.chartStateModel = chartStateModel;
+
+        chartStateModel.visualizationTypeProperty().addListener(
+                (ChangeListener<? super VisualizationType>) (dont, care, newVisualizationType) -> {
```

Comment:
sadly, java does not provide a keyword for ignored parameters :/

Reply:
Yet somewhere resolved to more confusing names that do not state, that these arguments are not used somewhere along the other commits.

Reply:
it doesn't matter if they are used or not, but it's important to clarify what they offer.
Maybe those variables are currently not used but this can change in later refinements.
So please change the code to:
`(ChangeListener<? super VisualizationType>) (source, oldVisualizationType, newVisualizationType) -> {`

Reply:
Well, it would clarify something that already is defined and documented on the changelistener (:
However, I actually changed that and do not know where my changes went.

Reply:
Jap, I was kind of confused as well, but I just figured it out what went wrong.
You fixed it in `PowerTerminalSidebarPaneController.java` instead :D

Reply:
checkout 62de25fb486b36198b5cd7da19b159e2b9b2d5eb

Reply:
Oh, rip. Will be done immediately.

(The other issues on which i didn't comment are either in the UnitMenu (asked svenja to fix that since its confusing me) or in the DBHandler (David should fix that, if he didnt already, maybe you can remind him tomorrow))

Reply:
Jap for sure, on my list :)